### ISSUE TITLE
Noob Question about .Net 4.8

### ISSUE THREAD

**Issue by gregpakes** [aspnet/RoslynCodeDomProvider#112]
I am targeting .net 4.8.  if I want to use c# 6+ features in my razor views, do I need this package?  

I am confused as to why I need this package if I am targeting 4.8.

**Comment by StephenMolloy** [comment]
TLDR - Yes, you probably need this package.

The C# compiler that ships "in box" with the .Net framework hasn't been updated with new features in years. The compiler team decided long ago that it would be better to ship out of box on a different cadence independent of the long lifecycle of the .Net Framework at that time. Even the csc.exe that comes "in box" with .Net 4.8 only supports up to C#5. So if you want C# 6+ features, you need to bring new compilers to your project. How you do that looks different depending on your environment and the type of project you are developing. If you're working on a non-web app in Visual Studio, then Visual Studio ships with and knows where to find the more modern compiler package and uses that to compile your app. If you're doing some on-the-fly compilation or code analysis of your own in your project, then you will need those updated compiler binaries too... which you get by including the Microsoft.Net.Compilers package in your project.

ASP.Net projects are different though. ASP.Net is responsible for compiling pages and handlers and controls and what-not into code files, and then also maintains responsibility for compiling those code files into binaries. And it often does this on the fly at runtime. Which is why you can deploy a site full of .aspx files to a server and get running code out of it. The way ASP.Net does this is through the CodeDom facilities of .Net. The default "in box" experience has those CodeDom utilities ultimately ending up at the "in box" csc.exe to go from the assembled .cs file to binary.

You mentioned razor views. While I believe razor can be used outside the context of a web application, the word "views" suggests to me that you're developing an ASP.Net MVC 5 project. In order for an ASP.Net project to get C# 6+ features, you do indeed need this project.
